src/controllers/queries/index.ts

```diff
@@ -51,13 +56,110 @@ export class QueriesController extends HTMLController {
     return this.html(this.queryManagementTemplates.listPage(queries, search))
   }
+
+  /**
+   * Retrieves the query page
+   */
+  @SuccessResponse(200)
+  @Get('/new/scope-3-carbon-consumption')
+  public async scope3CarbonConsumption(@Query() search?: string): Promise<HTML> {
+    const query = search ? [['company_name', 'ILIKE', `%${search}%`]] : {}
+    const connections = await this.db.get('connection', query, [['updated_at', 'desc']])
+    this.logger.debug('scope-3-carbon-consumption requested')
+    this.setHeader(
+      'HX-Replace-Url',
+      search
+        ? `/queries/new/scope-3-carbon-consumption?search=${encodeURIComponent(search)}`
+        : `/queries/new/scope-3-carbon-consumption`
+    )
+
+    return this.html(
+      this.scope3CarbonConsumptionTemplates.newScope3CarbonConsumptionFormPage('companySelect', connections, search, {
+        companyName: '',
+        companyNumber: '',
+      })
+    )
+  }
+
+  /**
+   * Retrieves the stage page
+   */
+  @SuccessResponse(200)
+  @Post('/new/scope-3-carbon-consumption/stage')
+  public async scope3CarbonConsumptionStage(
+    @Body()
+    body: {
+      companyNumber: COMPANY_NUMBER
+      companyName?: string
+      productId?: string
+      quantity?: number
+      action: 'companySelect' | 'form' | 'success'
+    }
+  ) {
+    const formStage: NewFormStage = body.action
+
+    if (formStage !== 'success') {
+      return this.html(
+        this.scope3CarbonConsumptionTemplates.newScope3CarbonConsumptionFormPage(formStage, [], '', {
+          companyName: '',
+          companyNumber: body.companyNumber,
+        })
+      )
+    }
+    const company = await this.db.get(
+      'connection',
+      [['company_number', 'ILIKE', `%${body.companyNumber}%`]],
+      [['updated_at', 'desc']]
+    )
+    if (!body.productId || !body.quantity) {
+      throw new Error('ProductId or quantity is missing.')
+    }
+    const newQueryId = await this.insertNewQuery(
+      company[0].id,
+      'Scope 3 Carbon Consumption',
+      `{ "productId": "${body.productId}", "quantity": "${body.quantity}" }`
+    )
+    if (!newQueryId) {
+      throw new Error('Query isertion was unsuccessful.')
+    }
+    //final stage
+    return this.html(
+      this.scope3CarbonConsumptionTemplates.newScope3CarbonConsumptionFormPage(formStage, [], '', {
+        companyName: company[0].company_name,
+        companyNumber: body.companyNumber,
+      })
+    )
+  }
+  private async insertNewQuery(
+    connection_id: string,
+    query_type: string,
+    details: string,
+    status: 'resolved' | 'pending_your_input' | 'pending_their_input' = 'pending_their_input'
+  ) {
+    try {
+      let queryId: string = ''
+
+      const [record] = await this.db.insert('query', {
+        connection_id: connection_id,
+        query_type: query_type,
+        status: status,
+        details: details,
+      })
+      queryId = record.id
+
+      return queryId
+    } catch (err) {
+      throw err
+    }
```

Comment:
why catch and rethrow, could remove try/catch? Unless something needs to happen with the error